src/modules/user-settings/hooks/use-editor-data.ts

```diff
@@ -3,15 +3,30 @@ import { getIpcApi } from 'src/lib/get-ipc-api';
 import { SupportedEditor } from 'src/modules/user-settings/lib/editor';
 
 export function useEditorData() {
-	const [ editor, setEditor ] = useState< SupportedEditor >( 'vscode' );
-	const [ savedEditorValue, setSavedEditorValue ] = useState< SupportedEditor >( 'vscode' );
+	const [ editor, setEditor ] = useState< SupportedEditor | string >( '' );
+	const [ savedEditorValue, setSavedEditorValue ] = useState< SupportedEditor | string >( '' );
```

Comment:
```suggestion
	const [ editor, setEditor ] = useState< SupportedEditor >();
	const [ savedEditorValue, setSavedEditorValue ] = useState< SupportedEditor >();
```
Initiating the state like so sets the variable types to `SupportedEditor | undefined`. 

We can please Typescript if we change `getSavedEditor` to default to return `undefined` and the two following functions like so:

```ts
	const handleEditorChange = ( newEditor: SupportedEditor | undefined ) => {
		if ( newEditor ) {
			setEditor( newEditor );
		}
	};

	const saveEditorPreference = async () => {
		if ( editor ) {
			await getIpcApi().saveUserEditor( editor );
		}
	};
```

The current type definitions force us to use the `as` operator all over the place, which is essentially short-circuiting type checking. Let's try to update the type definitions so we don't have to use the `as` operator.

Reply:
Thanks for highlighting it. I was literaly thinking about the change I made last evening related to using <string>. I am in the process of refactoring it.

Reply:
Nice 👍

Reply:
I factored the code to use `undefined`. Added in [1dcdca3](https://github.com/Automattic/studio/pull/1271/commits/1dcdca351bb2bc4bc31b18ae803479c0d48dc3e2)!